plugins/BEdita/Core/src/Model/Table/DateRangesTable.php

```diff
@@ -127,19 +138,139 @@ public function buildRules(RulesChecker $rules)
      */
     protected function findDateRanges(Query $query, array $options)
     {
-        $options = array_intersect_key($options, array_flip(['start_date', 'end_date']));
+        $allowed = array_flip([
+            'start_date',
+            'end_date',
+            'from_date',
+            'to_date'
+        ]);
+        $options = array_intersect_key($options, $allowed);
+        if (empty($options)) {
+            throw new BadFilterException([
+                'title' => __d('bedita', 'Invalid data'),
+                'detail' => 'start_date or end_date or date parameter missing',
+            ]);
+        }
+
+        $query = $this->fromToDateFilter($query, $options);
+        unset($options['from_date'], $options['to_date']);
+
+        // filter on `start_date` and `end_date` fields
         $options = array_combine(
             array_map([$this, 'aliasField'], array_keys($options)),
             array_values($options)
         );
 
-        if (empty($options)) {
+        return $this->fieldsFilter($query, $options);
+    }
+
+    /**
+     * Create Time object from $time string
+     *
+     * @param string $time Input time.
+     * @return null|\Cake\I18n\Time
+     */
+    protected function getTime(string $time): ?Time
+    {
+        if (empty($time)) {
+            return null;
+        }
+        try {
+            $res = new Time($time);
```

Comment:
minor: it could be `return new Time($time);` and then remove `return $res;` (lines 186-187)

Reply:
right, done